### PR TITLE
[18.01] Fix workflow step cloning

### PR DESCRIPTION
**Pull request by mvdbeek**
Problem one is that passing the tool_state as the input payload doesn't
preserve the tool state, adressed with the client-part fix.

Then passing the tool state still has no effect since the state is
properly recovered in `module_factory.from_dict`, but then overwritten
by recovering the state starting with an empty dict.  It looks like this
isn't necessary as we call populate_state anyway when calling
`module.get_config_form()`.

Fixes https://github.com/galaxyproject/galaxy/issues/5879.

### CODE REVIEW

**Comment by nsoranzo** [comment]
In #5879 you mentioned that this affects also 18.01 (and 18.05 very probably), do you want to change the target branch?

**Comment by mvdbeek** [comment]
Damnit, I guess I don't know how to rebuild the client nowadays :/

**Comment by dannon** [comment]
@mvdbeek For 18.01 it should be just `make client-production-maps`, which will do everything.

**Comment by mvdbeek** [comment]
Yep, that was it!

**Review by nsoranzo — APPROVED** [review]

File: lib/galaxy/webapps/galaxy/api/workflows.py [hunk 1]
```diff
@@ -496,9 +495,6 @@ def build_module(self, trans, payload={}):
         """
         inputs = payload.get('inputs', {})
         module = module_factory.from_dict(trans, payload)
-        module_state = {}
-        populate_state(trans, module.get_inputs(), inputs, module_state, check=False)
-        module.recover_state(module_state)
         return {
             'label'             : inputs.get('__label', ''),
             'annotation'        : inputs.get('__annotation', ''),
```
Comment (guerler) [inline lib/galaxy/webapps/galaxy/api/workflows.py]: Should this use `tool_state` instead of `inputs` to prevent it from being returned empty?

Comment (mvdbeek) [reply lib/galaxy/webapps/galaxy/api/workflows.py]: Ah, good point, let me try this.

Comment (mvdbeek) [reply lib/galaxy/webapps/galaxy/api/workflows.py]: Actually I see now that for new modules we do need to call `populate_state` here, not sure though why. Also I think this should stay `inputs.get(...)` as when you update the label or annotations this is being included in the `inputs`. 

It's just the cloning step that behaves different and directly copies label and annotation into the new node. So in practice copying labels / annotations works.

**Comment by jmchilton** [comment]
Seems odd that static/scripts/libs/vue.js is just added, is this a client build problem?

**Comment by dannon** [comment]
Could be cruft from switching between branches that got unintentionally
staged.

**Comment by mvdbeek** [comment]
Huh, I delete the file and rebuilt the client using `make client-production-maps`, it does put vue.js back in the working tree. Should I just back it out and ignore this ?

**Comment by mvdbeek** [comment]
For the record @dannon was right, just deleted `client/galaxy/scripts/lib` and then backed out the missing files with `git checkout client/galaxy/scripts/lib`.

**Pull request merged**